apps/server/src/services/userService.ts

```diff
@@ -0,0 +1,77 @@
+import { eq } from 'drizzle-orm';
+import { db } from '../clients/postgres';
+import { Users } from '../db/schema';
+import { EmailService } from '../utilities/emailService';
+import { hashString } from '../utilities/hashString';
+import { generateEncodedString } from '../utilities/generateEncodedString';
+import { verifyEncodedString } from '../utilities/verifyEncodedString';
+
+interface TUserRegisterBody {
+    fullName: string;
+    email: string;
+    password: string;
+    passwordConfirm: string;
+}
+
+const { JWT_AUTH_SECRET_KEY, JWT_AUTH_EXPIRES_IN, API_URL } = process.env;
+export class UserService {
+    constructor() {}
+
+    async register(requestBody: TUserRegisterBody): Promise<void> {
+        const { fullName, email, password, passwordConfirm } = requestBody;
+
+        const existingUser = await db
+            .select()
+            .from(Users)
+            .where(eq(Users.email, email!))
+            .limit(1);
+        if (existingUser.length)
+            throw Error('This user is already registered!');
+
+        if (password !== passwordConfirm)
```

Comment:
This check should be before querying the user. Plus do this inside zod schema with zod.superRefine to keep your service validation-free.

Reply:
This issue continues